github/release_notes/util.py

```diff
@@ -486,14 +442,31 @@ def release_note_block(
     category = code_block.get('category')
     target_group = code_block.get('target_group')
     source_repo = code_block.get('source_repo')
-    if source_repo:
+
+    if source_component:
+        reference_id = code_block.get('reference_id')
+        reference_type = reference_type_for_type_identifier(code_block.get('reference_type'))
+        user_login = code_block.get('user')
+    elif source_repo:
+        try:
+            # try to fetch cd for the parsed source repo. The actual version does not matter,
+            # we're only interested in the components GithubAccess (we assume it does not
+            # change).
+            source_cd = product.v2.greatest_component_version_by_name(
```

Comment:
in general, it is preferrable to use `cnudie.retrieve`, as this delegates to delivery-svc, if possible (and also does in-memory caching, which removes the need to handling a cache-dir in our (transient) use-case)